packages/quantic/force-app/main/default/lwc/quanticNotifications/__tests__/quanticNotifications.test.js

```diff
@@ -0,0 +1,189 @@
+/* eslint-disable no-import-assign */
+// @ts-ignore
+import QuanticNotifications from 'c/quanticNotifications';
+// @ts-ignore
+import {createElement} from 'lwc';
+import * as mockHeadlessLoader from 'c/quanticHeadlessLoader';
+import {AriaLiveRegion} from 'c/quanticUtils';
+
+jest.mock('c/quanticHeadlessLoader');
+jest.mock('c/quanticUtils');
+
+const exampleNotifications = ['notification1', 'notification2'];
+
+let notificationsState = {
+  notifications: exampleNotifications,
+};
+let isInitialized = false;
+
+const exampleEngine = {
+  id: 'mock engine',
+};
+
+const functionsMocks = {
+  buildNotifyTrigger: jest.fn(() => ({
+    state: notificationsState,
+    subscribe: functionsMocks.subscribe,
+  })),
+  dispatchMessage: jest.fn(() => {}),
+  subscribe: jest.fn((cb) => {
+    cb();
+    return functionsMocks.unsubscribe;
+  }),
+  unsubscribe: jest.fn(() => {}),
+};
+
+// @ts-ignore
+AriaLiveRegion.mockImplementation(() => {
+  return {
+    dispatchMessage: functionsMocks.dispatchMessage,
+  };
+});
```

Comment:
You are mocking the AriaLiveRegion implementation but you are not benefiting from that at all, 
a part of the unit tests should also cover the logic about the Aria live Region to confirm the right methods have been called with the right parameters.

Reply:
this was originally because otherwise, when we call update state, the tests would fail and give the following error:
`TypeError: Cannot read properties of undefined (reading 'dispatchMessage')`

Reply:
Actually you were getting the issue because the order of execution of the code was not correct, that's why I told you about the necessity of the change made in `quanticNotifications.js` offline that includes changing the order of two lines of code.
Mocking this is not what's fixing the issue but rather the change in order.

But that's not the point of my comment, my point is that as part of the unit tests we should test the logic that takes care of dispatching messages to aria live regions by confirming that the related methods are called in a proper way.

Reply:
added a test though

Reply:
If you also commented out `jest.mock('c/quanticUtils');` I bet you wouldn't get  any issue x)

Reply:
Because that's what determines if the real `AriaLiveRegion` method will be used or just mock function will be used instead of it.